pkg/bgpv1/agent/controller.go

```diff
@@ -144,6 +156,8 @@ type Controller struct {
 	// and provides a declarative API for configuring BGP peers.
 	BGPMgr BGPRouterManager
 
+	SecretsNamespace string
```

Comment:
Could you elaborate more about the reason for introducing this (and the Helm option)? I guess it is introduced to reduce the attack surface by limiting the namespace that can be accessed from BGP CPlane, but it's worth noting the reason for the broader users/developers.

Reply:
I've added a note in the docs. As you guessed it means the helm chart can safely set up default permissions, without potentially accidentally exposing other secrets in the cluster. It's also easier to just watch a namespace and this shouldn't change too often.